packages/neutrino/index.js

```diff
@@ -4,14 +4,21 @@ const webpack = require('./webpack');
 
 const IDENTITY = a => a;
 const configPrefix = 'neutrino.config';
+const nodeEnvsMode = ['production', 'development'];
 
 module.exports = (middleware = { use: ['.neutrinorc.js'] }, options = {}) => {
   const neutrino = new Neutrino(options);
   const args = yargs.parse(process.argv);
-  const mode = args.mode || 'production';
 
-  if (!process.env.NODE_ENV) {
+  let mode = args.mode;
+
+  if (mode) {
     process.env.NODE_ENV = mode;
+  } else if (process.env.NODE_ENV) {
+    // Development mode is most appropriate for a !production NODE_ENV (such as `NODE_ENV=test`).
+    mode = (process.env.NODE_ENV === 'production') ? 'production' : 'development';
+  } else {
+    throw new Error('Either --mode must be specified via the CLI, or else NODE_ENV be defined.');
```

Comment:
Alternatives to throwing are:
* default to `NODE_ENV='production'` and `mode='production'`
* default to `NODE_ENV='production'` and don't set `mode` at all. This would mean for workflows that include a webpack compile (ie not `yarn lint`), webpack would output the "no mode was specified, using production" warning message, which might help solve the "prevent people from accidentally omitting mode/NODE_ENV` concern.

Both of the alternatives would mean we could drop the `process.env.NODE_ENV = ...` boilerplate from the template `.eslintrc.js` files -- though we would still need it for `karma.config.js`.

Reply:
@eliperelman and this?